rustls-symcrypt/README.md

```diff
@@ -0,0 +1,120 @@
+# SymCrypt Provider for Rustls
+
+This crate provides integration for using `SymCrypt` cryptographic functionalities with the `rustls` crate, by implementing the required traits specified by `rustls`.
+
+**Note on Platform Support:** This crate is fully supported on `Windows AMD64` and `Linux Mariner` platforms. For `Ubuntu`, support is classified as partial. While `rustls-symcrypt` has undergone some testing on `Ubuntu`,
+full compatibility and performance cannot be guaranteed across all `Ubuntu` environments.
+
+**Notes:**
+- QUIC is not yet supported.
+- Integration with `rustls-cng` and `rustls-platform-verifier` are in progress.
+
+## Supported Ciphers
+
+Supported cipher suites are listed below, ordered by preference. IE: The default configuration prioritizes `TLS13_AES_256_GCM_SHA384` over `TLS13_AES_128_GCM_SHA256`.
+
+### TLS 1.3
+
+```ignore
+TLS13_AES_256_GCM_SHA384
+TLS13_AES_128_GCM_SHA256
+TLS13_CHACHA20_POLY1305_SHA256 // Enabled with the `chacha` feature
+```
+
+**Note:** `TLS13_CHACHA20_POLY1305_SHA256` is disabled by default. Enable the `chacha` feature in your `Cargo.toml` to use this cipher suite.
+
+### TLS 1.2
+
+```ignore
+TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 // Enabled with the `chacha` feature
+TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
+TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 // Enabled with the `chacha` feature
+```
+
+## Supported Key Exchanges
+
+Key exchanges are listed below, ordered by preference. IE: `SECP384R1` is preferred over `SECP256R1`.
+
+```ignore
+SECP384R1
+SECP256R1
+X25519 // Enabled with the `x25519` feature
+```
+
+**Note:** `X25519` is disabled by default. To enable, add `x25519` feature in your `Cargo.toml`.

```

Comment:
>X25519` is disabled by default. [](http://example.com/codeflow?start=11&length=31)

Why?

Reply:
Recommended & required under some compliance standards
------------------------------------------------------
NIST P-384 (also known as "NistP384" and "secp384r1")
NIST P-256 (also known as "NistP256", "secp256r1", and "prime256v1")